src/confd/src/ietf-system.c

```diff
@@ -720,35 +693,48 @@ static int sys_del_user(char *user)
 static int sys_add_new_user(sr_session_ctx_t *sess, char *name)
```

Comment:
There's quite a lot going on in this function now. Some ideas for improvements:
- Create a helper function that calls `adduser` either with or without explicit UID/GID. I.e. it would have a signature like `sys_call_adduser(sr_session_ctx_t *sess, const char *name, int uid, int gid)` where 0 could be used as sentinel value for both `uid` and `gid` since `root` will never be created. This could then also call `sys_find_usable_shell()`  to select an appropriate shell.
- Use `sys_del_user()` to remove stale user directories. This could also get rid of any SSH keys, which seems reasonable given that we've decided to nuke this user's data.
- With the two previous points in place, I think the `goto retry` falls away, which I had to read a few times before I convinced myself that there was no way of creating an infinite loop with it.

Reply:
Wow, thanks! I thought we could've had a chat about it this morning, but you just went ahead and smashed it.  I'll get right ahead with your suggestions! :smiley:

Reply:
Still a bit messy, but should be a lot more readable now with your suggestions in place.

Reply:
Well I guess you could say its messy from all of the log messages, but we want those! Otherwise it think it looks very straightforward now! :sparkles: